src/storage/data_object_storage_registry.rs

```diff
@@ -104,10 +108,9 @@ decl_module! {
         pub fn add_relationship(origin, cid: T::ContentId) {
             // Origin has to be a storage provider
             let who = ensure_signed(origin)?;
-            // TODO check for being staked as a storage provider
-            // if !T::Members::is_active_member(&who) {
-            //     return Err(MSG_CREATOR_MUST_BE_MEMBER);
-            // }
+
+            // Check that the origin is a storage provider
+            ensure!(<T as Trait>::Roles::account_has_role(&who, actors::Role::Storage), MSG_ONLY_STORAGE_PROVIDER_MAY_CREATE_DOSR);
```

Comment:
I don't think the cast to Trait is necessary here.

Reply:
I was wrong again.. :)

Reply:
I think its because the Roles type name is on multiple module traits, so the cast is needed to 'pick' the correct one. (disambiguation)